src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -965,8 +965,8 @@ void Transformations::MainSnippets(void) {
             size_t bf16Factor = 2;
             if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_amx)) {
                 const auto& b_shape = matmul->get_input_partial_shape(1);
-                const auto K = matmul->get_transpose_b() ? *b_shape.rbegin() : *(b_shape.rbegin()++);
-                const auto N = matmul->get_transpose_b() ? *(b_shape.rbegin()++) : *b_shape.rbegin();
+                const auto K = matmul->get_transpose_b() ? *b_shape.rbegin() : *(++b_shape.rbegin());
+                const auto N = matmul->get_transpose_b() ? *(++b_shape.rbegin()) : *b_shape.rbegin();
```

Comment:
Minor: redundant brackets (and on the prev line)
```suggestion
                const auto N = matmul->get_transpose_b() ? *++b_shape.rbegin() : *b_shape.rbegin();
```